graspologic/plot/plot_matrix.py

```diff
@@ -0,0 +1,933 @@
+
+    # make_axes_locatable allows us to create new axes like colormap
+    divider = make_axes_locatable(ax)
+
+    # draw separators
+    # draw top separators
+    draw_separators(
+        ax,
+        ax_type="x",
+        meta=col_meta,
+        group_class=col_group_class,
+        plot_type=plot_type,
+        gridline_kws=gridline_kws,
+    )
+    # draw left separators
+    draw_separators(
+        ax,
+        ax_type="y",
+        meta=row_meta,
+        group_class=row_group_class,
+        plot_type=plot_type,
+        gridline_kws=gridline_kws,
+    )
+
+    # draw highlighted separators
+    if highlight_kws is None:
+        highlight_kws = dict(color="black", linestyle="-", linewidth=1)
+    # draw top highlights
+    if col_highlight is not None:
+        draw_separators(
+            ax,
+            ax_type="x",
+            meta=col_meta,
+            group_class=col_highlight,
+            plot_type=plot_type,
+            gridline_kws=highlight_kws,
+        )
+    # draw left highlights
+    if row_highlight is not None:
+        draw_separators(
+            ax,
+            ax_type="y",
+            meta=row_meta,
+            group_class=row_highlight,
+            plot_type=plot_type,
+            gridline_kws=highlight_kws,
+        )
+
+    # these two variables are used to identify the first axes
+    col_first_ax = True
+    row_first_ax = True
+    # draw colors
+    # draw top colors
+    if len(col_color_class) > 0:
+        if col_color_pad is None:
+            col_color_pad = len(col_color_class) * [0.5]
+        if col_first_ax:
+            col_color_pad[0] = 0
+            col_first_ax = False
+
+        rev_color_class = list(col_color_class[::-1])
+        for i, sc in enumerate(rev_color_class):
+            color_ax = divider.append_axes(
+                "top", size="3%", pad=col_color_pad[i], sharex=ax
+            )
+            _remove_shared_ax(color_ax)
+            draw_colors(
+                color_ax,
+                meta=col_meta,
+                divider=divider,
+                ax_type="x",
+                color_class=rev_color_class[i],
+                palette=col_palette,
+            )
+    # draw left colors
+    if len(row_color_class) > 0:
+        if row_color_pad is None:
+            row_color_pad = len(row_color_class) * [0.5]
+        if row_first_ax:
+            row_color_pad[0] = 0
+            row_first_ax = False
+
+        rev_color_class = list(row_color_class[::-1])
+        for i, sc in enumerate(rev_color_class):
+            color_ax = divider.append_axes(
+                "left", size="3%", pad=row_color_pad[i], sharey=ax
+            )
+            _remove_shared_ax(color_ax)
+            draw_colors(
+                color_ax,
+                meta=row_meta,
+                divider=divider,
+                ax_type="y",
+                color_class=rev_color_class[i],
+                palette=row_palette,
+            )
+
+    # draw ticks
```

Comment:
It looks to me that the ticks are on or off for all classes. What if I had several classes but I only wanted to show ticks for some of them (maybe because some classes had too many categories to show or I had too many classes, etc)? Do you think we should accommodate cases like that as well?

Reply:
Right now the ticks are drawn as separate axis for all the group_class classes. If the user wants to sort the matrix by two classes, then there will be two tick axes, each separating the categories in each class.

Reply:
So it either shows both tick axes or does not show any?

Reply:
yes

Reply:
Right so I was asking if you think it would be useful to have the option of showing some, but not necessarily all, of the tick axes

Reply:
Yes I can do that. If users don't specify, then all tick axes will be shown, and if they do, then only the particular one will.

Reply:
@tliu68 isn't this what already happens if you make item_order a categorical variable? Please let me know if I'm not understanding. I do think we want this functionality but I think it is (mostly) taken care of by that.

Reply:
Ah right! I agree!